src/Sarif.Viewer.VisualStudio/LoadSarifLogService.cs

```diff
@@ -33,40 +28,76 @@ public void LoadSarifLog(string path, bool promptOnSchemaUpgrade = true)
         /// <inheritdoc/>
         public void LoadSarifLog(string path)
         {
-            // For now this is being done on the UI thread
-            // and is only required due to the message box being shown below.
-            // This will be addressed when https://github.com/microsoft/sarif-visualstudio-extension/issues/160
-            // is fixed.
-            ThreadHelper.ThrowIfNotOnUIThread();
+            if (string.IsNullOrWhiteSpace(path))
+            {
+                return;
+            }
 
             ErrorListService.ProcessLogFile(path, ToolFormat.None, promptOnLogConversions: true, cleanErrors: true);
         }
 
         /// <inheritdoc/>
         public void LoadSarifLogs(IEnumerable<string> paths)
         {
-            // For now this is being done on the UI thread
-            // and is only required due to the message box being shown below.
-            // This will be addressed when https://github.com/microsoft/sarif-visualstudio-extension/issues/160
-            // is fixed.
-            ThreadHelper.ThrowIfNotOnUIThread();
-
             this.LoadSarifLogs(paths, promptOnSchemaUpgrade: false);
         }
 
         /// <inheritdoc/>
         public void LoadSarifLogs(IEnumerable<string> paths, bool promptOnSchemaUpgrade)
         {
-            // For now this is being done on the UI thread
-            // and is only required due to the message box being shown below.
-            // This will be addressed when https://github.com/microsoft/sarif-visualstudio-extension/issues/160
-            // is fixed.
-            ThreadHelper.ThrowIfNotOnUIThread();
+            LoadSarifLogAsync(paths, promptOnSchemaUpgrade).FileAndForget("Microsoft/SARIF/Viewer/LoadSarifLogs");
```

Comment:
>FileAndForget("Microsoft/SARIF/Viewer/LoadSarifLogs") [](start = 60, length = 53)

What is this?

Reply:
When running a task and ignoring the result, this is an extension method (provided by VS) that captures any failures and raises telemetry events rather than potentially crashing the process.

---
In reply to: [494583877](https://github.com/microsoft/sarif-visualstudio-extension/pull/202#discussion_r494583877) [](ancestors = 494583877)

Reply:
Our extension actually does this so we can capture telemetry to our own app-insights key:

        /// <summary>
        /// Fires and forgets a task. The task will run in the background.
        /// </summary>
        /// <remarks>
        /// This leverages Visual Studio's Joinable Task Factory to "file and forget" (meaning log telemetry on failure)
        /// functionality.
        /// It is Visual Studio's preferred way to run tasks in the background where the result is discarded.
        /// It will prevent background tasks that throw exceptions from crashing visual Visual Studio and at
        /// the same time, allows telemetry to be logged.
        /// </remarks>
        /// <param name="task">The task for run.</param>
        /// <param name="faultEventName">An event name to use for VS's telemetry if the task faults.</param>
        /// <param name="faultDescription">A description to use if the task faults.</param>
        public static void FireAndForget(this System.Threading.Tasks.Task task, string faultEventName, string faultDescription)
        {
            task.FileAndForget(faultEventName, faultDescription, (exception) =>
            {
                var exceptionProperties = new Dictionary<string, string>
                {
                    { nameof(faultEventName), faultEventName },
                    { nameof(faultDescription), faultDescription },
                };

                ExtensionTelemetry.ReportException(exception, exceptionProperties);

                // Returning false here prevents Visual Studio from logging telemetry for this failure
                // which is desirable because we have our own telemetry.
                return false;
            });
        }


---
In reply to: [494964921](https://github.com/microsoft/sarif-visualstudio-extension/pull/202#discussion_r494964921) [](ancestors = 494964921,494583877)